cli/src/cluster_query.rs

```diff
@@ -394,6 +410,124 @@ pub fn process_get_slot(
     Ok(slot.to_string())
 }
 
+pub fn parse_show_block_production(matches: &ArgMatches<'_>) -> Result<CliCommandInfo, CliError> {
+    let epoch = value_t!(matches, "epoch", Epoch).ok();
+    let slot_limit = value_t!(matches, "slot_limit", u64).ok();
+
+    Ok(CliCommandInfo {
+        command: CliCommand::ShowBlockProduction { epoch, slot_limit },
+        require_keypair: false,
+    })
+}
+
+pub fn process_show_block_production(
+    rpc_client: &RpcClient,
+    epoch: Option<Epoch>,
+    slot_limit: Option<u64>,
+) -> ProcessResult {
+    let epoch_schedule = rpc_client.get_epoch_schedule()?;
+    let epoch_info = rpc_client.get_epoch_info_with_commitment(CommitmentConfig::max())?;
+
+    let epoch = epoch.unwrap_or(epoch_info.epoch);
+
+    let first_slot = epoch_schedule.get_first_slot_in_epoch(epoch);
+    let last_slot = std::cmp::min(
+        epoch_info.absolute_slot,
+        epoch_schedule.get_last_slot_in_epoch(epoch),
+    );
+
+    let first_slot = std::cmp::max(
+        last_slot.saturating_sub(slot_limit.unwrap_or(first_slot)),
+        first_slot,
+    );
+
+    let progress_bar = new_spinner_progress_bar();
+    progress_bar.set_message("Connecting...");
+    progress_bar.set_message(&format!("Fetching leader schedule for epoch {}...", epoch));
+
+    let leader_schedule = rpc_client
+        .get_leader_schedule_with_commitment(Some(first_slot), CommitmentConfig::max())?;
+
+    if leader_schedule.is_none() {
+        return Err(format!("Unable to fetch leader schedule for slot {}", first_slot).into());
+    }
+    let leader_schedule = leader_schedule.unwrap();
+    let total_slots = last_slot - first_slot;
+    assert!(total_slots <= leader_schedule.len() as u64);
+
+    let mut total_blocks_produced = 0;
+    let mut total_slots_missed = 0;
+    let mut leader_slot_count = HashMap::new();
+    let mut leader_missed_slots = HashMap::new();
+
+    for slot_index in 0..total_slots {
```

Comment:
This super slow loop mostly gets replaced by `rpc_client.get_confirmed_blocks(first_slot, last_slot)` once it exists!  Calling `rpc_client.get_confirmed_block()` on each slot just to figure out if it's confirmed or not is pretty 🤢

Reply:
I guess I better plumb getConfirmedBlocks into rpc_client ;)
I'll do that as follow-up pr, if #7550 looks good

Reply:
#7560